SADEAFxSMU/sadeaf-app
# The Majestic Monolith

Removed `sadeaf-api` and `sadeaf-worker` and moved all the code into web as part of the `serverMiddleware`. Deploy as a monolith to simplify the deployment packages and greatly reduce the spread of services required and therefore cost.

Related Issues: #38 #39 #40

# Caveats
- Race condition, web hook endpoint might not be up and hasura trigger it. (Not relevant because hasura has build in retry system.)
- Authentication not yet, doing in my next PR.
- Monolith yeah!
- I hate this design but it's the easier to understand and for them to take over. (Please comment on this if any)

# How has this been tested?
> I have removed worker test cases for brevity, so that I can create this PR faster.

```sh
cd sadeaf-web
yarn run dev
```

- Visit: http://localhost:3000
- Visit: http://localhost:3000/api/v1/accounts
- Visit: http://localhost:3000/api/v1/graphql
- Open PostMan or similar
- POST: http://localhost:3000/api/v1/hasura/webhook/email-example
```json 
{
    "from": "sadeaf.from@fuxing.io",
    "to": "123.sadeaf.to@fuxing.io",
    "subject": "TITLE",
    "body": "BODY"
}
```
- Check: http://localhost:18025
